File: app/src/main/java/com/AbdelAdil/myapplication/data/models/Product.kt
package com.AbdelAdil.myapplication.data.models

import com.google.gson.annotations.SerializedName

data class Product(
    val id: Int,
    val nom: String,
    val description: String,
    val prix: Double,
    val stock: Int,
    val images: List<String>? = null,
    val statut: String? = null,
    val categorie: Category? = null,
    val vendeur: User? = null,
    @SerializedName("date_creation") val dateCreation: String? = null
)

File: MyApplicationbackend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Product extends Model
{
    /** @use HasFactory<\Database\Factories\ProductFactory> */
    use HasFactory;

    protected $fillable = [
        'nom',
        'description',
        'prix',
        'stock',
        'images',
        'statut',
        'vendeur_id',
        'categorie_id',
    ];

    protected function casts(): array
    {
        return [
            'prix' => 'decimal:2',
            'stock' => 'integer',
            'images' => 'array',
        ];
    }

    public function vendeur(): BelongsTo
    {
        return $this->belongsTo(User::class, 'vendeur_id');
    }

    public function categorie(): BelongsTo
    {
        return $this->belongsTo(Category::class, 'categorie_id');
    }
}
